File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.java
import java.util.*;

class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        String answer = "Yes";
        int[] arr = new int[cards1.length];
        int[] arr2 = new int[cards2.length];
        for(int i = 0; i < arr.length; i++){
            arr[i] = Arrays.asList(goal).indexOf(cards1[i]);
            if(arr[i] != -1){
                for(int j = 0; j < i; j++){
                    if(arr[j] == -1){
                        answer = "No";
                        break;
                    }
                    if(arr[i] < arr[j]){
                        answer = "No";
                        break;
                    }
                }
            }
        }
        for(int i = 0; i < arr2.length; i++){
            arr2[i] = Arrays.asList(goal).indexOf(cards2[i]);
            if(arr2[i] != -1){
                for(int j = 0; j < i; j++){
                    if(arr2[j] == -1){
                        answer = "No";
                        break;
                    }
                    if(arr2[i] < arr2[j]){
                        answer = "No";
                        break;
                    }
                }
            }
        }
        
        
        return answer;
    }
}

File: 프로그래머스/1/159994. 카드 뭉치/카드 뭉치.swift
import Foundation

func isTrue(_ cards:[String], _ goal:[String]) -> Bool {
    var result = true
    var idxArr = Array<Int>()
    outer : for i in 0..<cards.count {
        if let idx = goal.firstIndex(of: cards[i]) {
            idxArr.append(idx)
        } else {
            idxArr.append(-1)
        }
        if idxArr[i] != -1 {
            for j in 0..<i {
                if idxArr[j] == -1 {
                    result = false
                    break outer
                }
                if idxArr[i] < idxArr[j] {
                    result = false
                    break outer
                }
            }
        }
    }
    return result
}

func solution(_ cards1:[String], _ cards2:[String], _ goal:[String]) -> String {
    var result = "Yes"
    var result1 = isTrue(cards1, goal)
    var result2 = isTrue(cards2, goal)
    if result1 == false || result2 == false {
        result = "No"
    }
    return result
}